wgsl/index.bs

```diff
@@ -931,9 +941,230 @@ a result value, such as the non-result-value effects of its subexpressions.
 
 TODO: example: non-result-value effect is any side effect of a function call subexpression.
 
+## Ideal Types ## {#ideal-types-sec}
```

Comment:
Maybe add to the note in the types overview about unwritten types to include ideal types and link here.